app/lib/common/utilities/optional.dart

```diff
@@ -0,0 +1,14 @@
+import 'package:flutter/foundation.dart';
+
+/// Inspired by Java's Optional type.
```

Comment:
[Java's optional](https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html). I don't have this for every type, not sure if overkill as empty string will suffice.